proposals/lip-0022.md

```diff
@@ -177,25 +177,19 @@ Internally, the algorithm works as follows:
+*   The round length is now 103 block slots. Currently, this is defined by the constant  <code>[ACTIVE_DELEGATES](https://github.com/LiskHQ/lisk-sdk/blob/310283408102ff9f891830225afe8124e833a608/config/default/constants.js#L55)</code>.  For clarity, this constant can be replaced for three constants defining the number of the top 101 delegates by `delegateWeight`, the number of standby delegates, i.e., 2, and the length of the round (sum of the two previous constants).
 *   The change in the length of the round affects the logic to check when a round is finished which is 103 block slots in total.
-*   `delegate1` and `delegate2` have to be included in the delegate list<sup>3</sup> before it is re-ordered for `round` + 1. Currently, the function <code>[getKeysSortByVote](https://github.com/LiskHQ/lisk-sdk/blob/7c31c9d4b12a72f18a35ead79198b2ba6d4bd091/framework/src/modules/chain/submodules/delegates.js#L32)</code> computes this list before being re-ordered.
+*   `delegate1` and `delegate2` have to be included in the delegate list<sup>3</sup> before it is re-ordered for `round` + 1. Currently, the function <code>[createRoundDelegateList](https://github.com/LiskHQ/lisk-sdk/blob/5997714fde5181701fa0508eebd7f3c595cbf92c/elements/lisk-dpos/src/delegates_list.ts#L148)</code> computes this list before being re-ordered.
 
-Bear in mind that this proposal does not change any other point in the round processing once the list with the 103 forging delegates for the next round is computed.  
+This LIP also proposes to use the 20 bytes adddress of a delegate as defined in [LIP-0018](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0018.md) for the [delegate lists](https://github.com/LiskHQ/lisk-sdk/blob/5997714fde5181701fa0508eebd7f3c595cbf92c/elements/lisk-dpos/src/delegates_list.ts#L124). This also implies two things:
 
-#### Delegate List Ordering
+* The delegate address is used for tie break instead of the public key when two delegates have the same `delegateWeight`.
 
-This proposal also updates the proposed changes in [LIP-0003](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0003.md). The variable `seedSource`, defined in the first bullet point of the Specification section in that document, is redefined as:
-
-```js
-var seedSource = randomSeed1;
-```
-
-The remaining changes of LIP-0003 remain the same and stay valid.
+* The delegate address has to be derived from the block property `generatorPublicKey` in order to verify the correct block slot (this is currently implemented in the function [verifyBlockForger](https://github.com/LiskHQ/lisk-sdk/blob/5997714fde5181701fa0508eebd7f3c595cbf92c/elements/lisk-dpos/src/delegates_list.ts#L227)).
```

Comment:
Isn't that how we do it anyways ?

Reply:
not really. Now the delegate lists contains the delegates' public keys and this can be directly verified against the property `generatorPublicKey` of the corresponding block. Once these changes are active, one will need to derive an address (as it is done now or as it is specified in LIP 0018) from `generatorPublicKey` in order to validate that the delegate who forged the block was the correct one for the corresponding slot.